src/Document.php

```diff
@@ -163,15 +163,30 @@ public function graphql_mutation_insert( $post_object, $filtered_input, $input,
 	 * @param  array $request_context An array containing the both body and query params
 	 * @return string Updated $parsed_body_params Request parameters.
 	 */
-	public function graphql_query_contains_queryid_cb( $parsed_body_params, $request_context ) {
-		if ( isset( $parsed_body_params['query'] ) && isset( $parsed_body_params['queryId'] ) ) {
+	public function graphql_query_contains_query_id_cb( $parsed_body_params, $request_context ) {
+
+		// if both query and queryId are set
+		// we should attempt to save the query document (per APQ)
+		if ( ! empty( $parsed_body_params['query'] ) && ! empty( $parsed_body_params['queryId'] ) ) {
 			// save the query
 			// The query string already coming from 'graphql_request_data' already has the query string unslashed.
 			$this->save( $parsed_body_params['queryId'], $parsed_body_params['query'] );
 
 			// remove it from process body params so graphql-php operation proceeds without conflict.
 			unset( $parsed_body_params['query'] );
 		}
+
+		// if the query is empty, but the queryId is set
+		if ( empty( $parsed_body_params['query'] ) && ! empty( $parsed_body_params['queryId'] ) ) {
+			$query_id     = $parsed_body_params['queryId'] ?: null;
```

Comment:
You already did the `! empty($parsed_body_params['queryId'])` do you need the ternary `?:` here?